client/src/js/services/FilterService.js

```diff
@@ -4,50 +4,183 @@ angular.module('bhima.services')
+  FilterList.prototype.registerDefaultFilters = function registerDefaultFilters(filterDefinitions) {
+    var formattedFilters = filterDefinitions.map(function (filterDefinition) {
+      var filter = new Filter(filterDefinition.key, filterDefinition.label, filterDefinition.valueFilter);
+      filter.setDefault(true);
+
+      if (filterDefinition.defaultValue) {
+        filter.assign(filterDefinition.defaultValue);
+      }
+      return filter;
+    });
+
+    // udpate index
+    this._indexList(this._filterIndex, formattedFilters);
+    this._defaultFilters = this._defaultFilters.concat(formattedFilters);
+  };
+
+  FilterList.prototype.registerCustomFilters = function registerCustomFilters(filterDefinitions) {
+    var formattedFilters = filterDefinitions.map(function (filterDefinition) {
+      var filter = new Filter(filterDefinition.key, filterDefinition.label, filterDefinition.valueFilter);
+      filter.setDefault(false);
+      return filter;
+    });
+
+    // udpate index
+    this._indexList(this._filterIndex, formattedFilters);
+    this._customFilters = this._customFilters.concat(formattedFilters);
+  };
+
+
+  // assigns the value of a filter, a filter with a value will be actively used
+  // during the HTTP/ UI export process
+  FilterList.prototype.assignFilter = function assignFilter(key, value, displayValue) {
+    this._filterIndex[key].assign(value, displayValue);
+  };
+
+  // accepts an array of key : filterValue objects that are assigned
+  // [
+  // { key : value },
+  // { key : value }
+  // ]
+  FilterList.prototype.assignFilters = function assignFilters(valueList) {
+    valueList.forEach(function (valueMap) {
+      this.assignFilter(valueMap.key, valueMap.value, valueMap.displayValue);
+    }.bind(this));
+  };
+
+  // alias for `assignFilters`, clears the currently active filters before
+  // calling the erferenced method
+  FilterList.prototype.replaceFilters = function replaceFilters(valueList) {
+    this._resetCustomFilters();
+    this.assignFilters(valueList);
+  };
+
+  // return filters for the view - this method will always be compatible with the bhFilter component
+  FilterList.prototype.formatView = function formatView() {
+    var activeFilters = this._filterActiveFilters();
+    var activeKeys = activeFilters.map(function (filter) {  return filter._key });
+
+    function keysInActive(filter) { return activeKeys.includes(filter._key); }
+
+    // parse into two lists
+    return {
+      defaultFilters : this._defaultFilters.filter(keysInActive),
+      customFilters : this._customFilters.filter(keysInActive)
+    };
+  };
+
+  // format filters for the server
+  // sendClientTimestamp - this will send an attribute hidden to the user
+  // returns a JSON object with active filters
+  FilterList.prototype.formatHTTP = function formatHTTP(sendClientTimestamp) {
+    var clientTimestamp = angular.isDefined(sendClientTimestamp) ? sendClientTimestamp : false;
```

Comment:
It would be clearer to name this `hasClientTimestamp` or `needsClientTimestamp` or something similar to indicate this is a boolean and not an actual timestamp.